### PR TITLE
ROC-2874: Add a developer tool to create a generic claim

### PR DESCRIPTION
**Pull request by jacmurphy**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-2874

### Change description
Added a button in test support to create a generic claim draft and directs you to the check and send page.

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/testing-support/routes/create-draft.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      res.render(Paths.createDraftPage.associatedView)
+    })
+  )
+  .post(Paths.createDraftPage.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response) => {
+      const user: User = res.locals.user
+      const drafts = await draftService.find(getDraftType(req), '100', user.bearerToken, (value) => value)
+
+      drafts.forEach(async draft => {
+        await new DraftService().delete(draft.id, user.bearerToken)
```
Comment (jacmurphy) [inline src/main/features/testing-support/routes/create-draft.ts]: This ts file uses much of the same code as `delete-drafts` to potentially delete any pre-existing claim drafts, to be replaced by the generic template.
If there is a better way to implement this more generically, please comment

File: src/main/features/testing-support/views/create-draft.njk [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{% extends "layout.njk" %}
+{% from "form.njk" import linkButton, csrfProtection, errorSummary %}
+{% from "date.njk" import dateInput %}
+
+{% set title = 'Create Draft | Testing support' %}
+{% set heading = t('Create Draft') %}
```
Comment (jacmurphy) [inline src/main/features/testing-support/views/create-draft.njk]: Personally i don't like having the same content for the header and the page. If anyone has a better suggestion, please comment

Comment (singh-surjeet) [inline src/main/features/testing-support/views/create-draft.njk]: @geek-soft Do we require translation for testing support pages i.e. t('Create Draft')

Comment (geek-soft) [reply src/main/features/testing-support/views/create-draft.njk]: Not really, testing support is just for us however I see we already started wrapping wording with `t` function on other testing support screens.

File: src/test/data/draft/createDraft.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import { individualDetails } from './partyDetails'
```
Comment (geek-soft) [inline src/test/data/draft/createDraft.ts]: File name `src/test/data/draft/createDraft.ts` is wrong, you do not export `createDraft` function here, you export sample JSON object. Please rename to something like `claimDraft`.

There is also `data/draft/claimDraft.ts` already there, any reason you did not decide to use that one?

REQ

File: src/main/features/testing-support/views/create-draft.njk (hunk 2 above)
Comment (jacmurphy) [reply src/main/features/testing-support/views/create-draft.njk]: 👍

Comment (jacmurphy) [reply src/main/features/testing-support/views/create-draft.njk]: Removed translation

File: src/main/features/testing-support/routes/create-draft.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const draftService = new DraftService()
+
+function getDraftType (req: express.Request): string {
+  return Object.keys(req.body.action)[0]
+}
+
+/* tslint:disable:no-default-export */
+export default express.Router()
+  .get(Paths.createDraftPage.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response) => {
```
[the remark this answers is not in the document]
Comment (jacmurphy) [reply src/main/features/testing-support/routes/create-draft.ts]: Without the async function, i cant loop through the drafts to delete them. I used the same code as the delete-draft.ts file

File: src/main/features/testing-support/routes/create-draft.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import * as express from 'express'
+
+import { Paths } from 'testing-support/paths'
+import { Paths as DraftSubmittion } from 'claim/paths'
```
[the remark this answers is not in the document]
Comment (jacmurphy) [reply src/main/features/testing-support/routes/create-draft.ts]: 👍

File: src/main/features/testing-support/routes/create-draft.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (geek-soft) [reply src/main/features/testing-support/routes/create-draft.ts]: @jacmurphy you are not looping through anything in GET handler.

File: src/main/features/testing-support/routes/create-draft.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+export default express.Router()
+  .get(Paths.createDraftPage.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response) => {
+      res.render(Paths.createDraftPage.associatedView)
+    })
+  )
+  .post(Paths.createDraftPage.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response) => {
+      const user: User = res.locals.user
+      const drafts = await draftService.find(getDraftType(req), '100', user.bearerToken, (value) => value)
```
[the remark this answers is not in the document]
Comment (jacmurphy) [reply src/main/features/testing-support/routes/create-draft.ts]: 👍

File: src/main/features/testing-support/routes/create-draft.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jacmurphy) [reply src/main/features/testing-support/routes/create-draft.ts]: Im reading a different comment from yours that addresses this.
👍

File: src/main/features/testing-support/views/create-claim-draft.njk [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+{% set heading = 'Create Claim Draft' %}
+
+{% block content %}
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <form method="post" novalidate>
+        {{ csrfProtection(csrf) }}
+
+        <div class="notice">
+          {{ linkButton('Create Claim Draft', 'action[claim]', 'button button-secondary') }}
```
Comment (geek-soft) [inline src/main/features/testing-support/views/create-claim-draft.njk]: A 'action[claim]' is no longer needed since you not using it to read draft type.

Comment (jacmurphy) [reply src/main/features/testing-support/views/create-claim-draft.njk]: removed 👍

File: src/test/data/draft/partyDetails.ts [hunk 8]
```diff
@@ -79,15 +79,15 @@ export const organisationDetails = {
...
+    postcode: 'SW1A 2AA'
   },
   hasCorrespondenceAddress: true,
   correspondenceAddress: {
     line1: 'Flat 40',
     line2: 'Street 40',
     line3: 'Cooler house name',
     city: 'Belfast',
-    postcode: 'B40A'
+    postcode: 'SW1A 1AA'
```
Comment (timja-kainos) [inline src/test/data/draft/partyDetails.ts]: london postcode but belfast city?

File: src/test/data/entity/party.ts [hunk 9]
```diff
@@ -75,14 +75,14 @@ export const organisation = {
...
-    postcode: 'E4A'
+    postcode: 'SW1A 2AA'
   },
   correspondenceAddress: {
     line1: 'Flat 40',
     line2: 'Street 40',
     line3: 'Cooler house name',
     city: 'Belfast',
-    postcode: 'B40A'
+    postcode: 'SW1A 1AA'
```
Comment (timja-kainos) [inline src/test/data/entity/party.ts]: same here?

File: src/main/features/testing-support/views/create-claim-draft.njk [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+{% set heading = 'Create Claim Draft' %}
+
+{% block content %}
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <form method="post" novalidate>
+        {{ csrfProtection(csrf) }}
+
+        <div class="notice">
+          {{ linkButton('Create Claim Draft', 'claim', 'button button-secondary') }}
```
Comment (timja-kainos) [inline src/main/features/testing-support/views/create-claim-draft.njk]: why use a linkButton? shouldn't this be an actual button?

**Pull request merged**